src/compiler/checker.ts

```diff
@@ -1662,6 +1663,13 @@ namespace ts {
             return type;
         }
 
+        function createNonPrimitiveType(): ResolvedType {
+            const type = setStructuredTypeMembers(

```

Comment:
@ahejlsberg can judge better, but I don't think you want to set any structured type members - I think you want to just need to add a branch in `getApparentType` for `object`.